app/assets/javascripts/govuk_publishing_components/lib/govspeak/youtube-link-enhancement.js

```diff
@@ -1,50 +1,101 @@
+    for (var i = 0; i < $youtubeLinks.length; ++i) {
+      var $link = $youtubeLinks[i]
       // if users have disabled 'campaigns' cookie in the new cookie page settings
       // we also need to disable the youtube video embed
       if (_this.hasDisabledEmbed($link) || !YoutubeLinkEnhancement.campaignCookies()) {
         return true
       }
-
-      var videoId = YoutubeLinkEnhancement.parseVideoId($link.attr('href'))
+      // insert script only once, but only if a video in the list is set to embed
+      // and campaign cookies are enabled
+      if (!apiScriptInserted) {
+        _this.insertApiScript()
+        apiScriptInserted = true
+      }
+      var videoId = YoutubeLinkEnhancement.parseVideoId($link.getAttribute('href'))
       if (videoId) {
-        _this.embedVideo($link, videoId)
+        _this.setupVideo($link, videoId)
       }
-    })
+    }
   }
 
-  YoutubeLinkEnhancement.prototype.hasDisabledEmbed = function ($link) {
-    return $link.attr('data-youtube-player') === 'off'
+  YoutubeLinkEnhancement.prototype.playerApiReady = function () {
+    // youtube iframe api is very pick about knowing when it's ready,
+    // so they want this function to be on the window object
+    window.onYouTubePlayerAPIReady = function () {
+      // youtube API provides the variable when it is definitely ready
+      YT = window.YT
+      for (var i = 0; i < this.playList.length; i++) {
+        var currentPlayer = this.createVideoPlayer(this.playList[i])
+        this.players[i] = currentPlayer
+      }
+    }.bind(this)
   }
 
-  YoutubeLinkEnhancement.prototype.embedVideo = function ($link, videoId) {
-    var $mediaPlayer = $('<span class="media-player" />')
+  YoutubeLinkEnhancement.prototype.insertApiScript = function () {
+    var tag = document.createElement('script')
+    tag.src = 'https:///www.youtube.com/iframe_api'
+    var firstScriptTag = document.getElementsByTagName('script')[0]
+    firstScriptTag.parentNode.insertBefore(tag, firstScriptTag)
 
-    $link.parent().replaceWith($mediaPlayer)
+    this.playerApiReady()
+  }
 
-    var protocol = this.protocol()
+  YoutubeLinkEnhancement.prototype.hasDisabledEmbed = function ($link) {
+    return $link.getAttribute('data-youtube-player') === 'off'
+  }
+
+  YoutubeLinkEnhancement.prototype.setupVideo = function ($link, videoId) {
+    var parent = $link.parentNode
+    var youtubeVideoMarkup = '<div class="youtube-video-container"><span id="' + videoId + '"></span></div>'
+    parent.innerHTML = youtubeVideoMarkup
+    this.playList.push({
+      id: videoId,
+      videoId: videoId,
+      title: $link.textContent
+    })
+  }
 
-    $mediaPlayer.player({
-      id: YoutubeLinkEnhancement.nextId(),
-      media: videoId,
-      captions: this.captions($link),
-      url: protocol + '//www.youtube.com/apiplayer?enablejsapi=1&version=3&playerapiid='
+  YoutubeLinkEnhancement.prototype.createVideoPlayer = function (playlistItem) {
+    return new YT.Player(playlistItem.id, {
```

Comment:
It might be clearer to have this reference window.YT and then remove the line from https://github.com/alphagov/govuk_publishing_components/pull/908/files#diff-b4722dbdd6fe718214588e4addc706d4R46 (which I found rather confusing)